### PR TITLE
Hw6

### PR DESCRIPTION
**Pull request by MaratMukhutdinov**

### CODE REVIEW

File: ru/mukhutdinov/Homework6/Calculator.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    private static double calculation;
+    private static Scanner scan = new Scanner(System.in);
+    private static String nextCalc;
+    private static Double newNumber;
+
+    public static void main(String[] args) {
+        System.out.println("Доступны следующие действия над числом: Сложение(+), Вычитание(-), Умножение(*), Деление(/) и Нахождение процента от числа(%)");
+        System.out.println("Если захотите завершить программу, введите знак *!* вместо знака операции");
+        mainMenu();
+
```
Comment (BulatMukhutdinov) [inline ru/mukhutdinov/Homework6/Calculator.java]: empty line

File: ru/mukhutdinov/Homework6/Calculator.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    private static void mainMenu() {
```
Comment (BulatMukhutdinov) [inline ru/mukhutdinov/Homework6/Calculator.java]: should be verb

File: ru/mukhutdinov/Homework6/Calculator.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+/* Задача: Реализовать класс Calculator, который будет содержать статические методы для операций вычитания, сложения,
+        умножения, деления и взятия процента от числа. Класс должен работать как с целыми числами, так и с дробями. */
+
+import java.util.Scanner;
+
+public class Calculator {
+    private static double calculation;
+    private static Scanner scan = new Scanner(System.in);
+    private static String nextCalc;
+    private static Double newNumber;
```
Comment (BulatMukhutdinov) [inline ru/mukhutdinov/Homework6/Calculator.java]: nextDouble?

File: ru/mukhutdinov/Homework6/Calculator.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+/* Задача: Реализовать класс Calculator, который будет содержать статические методы для операций вычитания, сложения,
+        умножения, деления и взятия процента от числа. Класс должен работать как с целыми числами, так и с дробями. */
+
+import java.util.Scanner;
+
+public class Calculator {
+    private static double calculation;
+    private static Scanner scan = new Scanner(System.in);
+    private static String nextCalc;
```
Comment (BulatMukhutdinov) [inline ru/mukhutdinov/Homework6/Calculator.java]: does it used?

File: ru/mukhutdinov/Homework6/Calculator.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
+package ru.mukhutdinov.Homework6;
+
+/* Задача: Реализовать класс Calculator, который будет содержать статические методы для операций вычитания, сложения,
+        умножения, деления и взятия процента от числа. Класс должен работать как с целыми числами, так и с дробями. */
+
+import java.util.Scanner;
+
+public class Calculator {
+    private static double calculation;
```
Comment (BulatMukhutdinov) [inline ru/mukhutdinov/Homework6/Calculator.java]: rename. Probably currentValue or currentNumber or something similar

File: ru/mukhutdinov/Homework6/Calculator.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        System.out.println("Если захотите завершить программу, введите знак *!* вместо знака операции");
+        mainMenu();
+
+    }
+
+    private static void mainMenu() {
+        for (; ; ) {
+            System.out.println("Ваше число на данный момент: " + calculation);
+            System.out.print("Введите число для выполнения данной операции: ");
+            newNumber = scan.nextDouble();
```
Comment (BulatMukhutdinov) [inline ru/mukhutdinov/Homework6/Calculator.java]: what would happened if I input 10000000000000000000000?
Prefer read line and parse it to BigDouble. Here on you can use 
```
BigDouble bd = new BigDouble(readLine);
bd.add(anotherBigDouble)
```

File: ru/mukhutdinov/Homework6/ObjectCounter.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    }
+}
+
+class objectStorage {
+    static int count = 0;
+    objectStorage() {
+        count++;
+    }
+}
+
```
Comment (BulatMukhutdinov) [inline ru/mukhutdinov/Homework6/ObjectCounter.java]: empty lines

File: ru/mukhutdinov/Homework6/ObjectCounter.java [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    public static void main(String[] args) {
+        objectStorage firstObject = new objectStorage();
+        objectStorage secondObject = new objectStorage();
+        objectStorage thirdObject = new objectStorage();
+        System.out.println("Создано объектов: " + objectStorage.count);
+    }
+}
+
+class objectStorage {
```
Comment (BulatMukhutdinov) [inline ru/mukhutdinov/Homework6/ObjectCounter.java]: Awful... follow naming conventions

File: ru/mukhutdinov/Homework6/ObjectCounter.java [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+    static int count = 0;
```
Comment (BulatMukhutdinov) [inline ru/mukhutdinov/Homework6/ObjectCounter.java]: prefer making it private and create getter method

File: ru/mukhutdinov/Homework6/Calculator.java (hunk 1 above)
Comment (MaratMukhutdinov) [reply ru/mukhutdinov/Homework6/Calculator.java]: Removed

File: ru/mukhutdinov/Homework6/Calculator.java (hunk 4 above)
Comment (MaratMukhutdinov) [reply ru/mukhutdinov/Homework6/Calculator.java]: removed

File: ru/mukhutdinov/Homework6/ObjectCounter.java (hunk 7 above)
Comment (MaratMukhutdinov) [reply ru/mukhutdinov/Homework6/ObjectCounter.java]: removed

File: ru/mukhutdinov/Homework6/ObjectCounter.java (hunk 8 above)
Comment (MaratMukhutdinov) [reply ru/mukhutdinov/Homework6/ObjectCounter.java]: Whoops

File: ru/mukhutdinov/Homework6/Calculator.java (hunk 6 above)
Comment (MaratMukhutdinov) [reply ru/mukhutdinov/Homework6/Calculator.java]: Done

File: ru/mukhutdinov/Homework6/ObjectCounter.java (hunk 9 above)
Comment (MaratMukhutdinov) [reply ru/mukhutdinov/Homework6/ObjectCounter.java]: Done

**Pull request merged**